### PR TITLE
NAS-111036 / 21.06-BETA.1 / Added defaults for 'aclmode' based on 'acltype' & warning shown

### PR DESCRIPTION
**Pull request by RehanY147**
NAS-111036

### CODE REVIEW

File: src/assets/i18n/af.json [hunk 1]
```diff
@@ -634,6 +634,7 @@
   "Change log": "",
   "Changes Saved": "",
   "Changes to <em>Hosts Allow</em> or <em>Hosts Deny</em> take effect when the  SMB service restarts.": "",
+  "Changes to ACL type affect how on-disk ZFS ACL is  written and read. When the ACL type is changed from POSIX to NFSv4,  no migration is performed for default and access ACLs encoded in the  posix1e acl extended attributes to native ZFS ACLs. When ACL type is  changed from NFSv4 to POSIX, native ZFS ACLs are not converted to  posix1e extended attributes, but the native ACL will be used internally  by ZFS for access checks. This means that the user must manually set  new ACLs recursively on the dataset after ACL type changes in order to  avoid unexpected permissions behavior. This action will be destructive,  and so it is advised to take a ZFS snapshot of the dataset prior to ACL  type changes and permissions modifications.": "",
```
Comment (undsoft) [inline src/assets/i18n/af.json:637]: Can we add new lines somewhere? It's hard to read one large block of text.

File: src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts [hunk 2]
```diff
@@ -666,6 +655,37 @@ export class DatasetFormComponent implements FormConfiguration {
...
+            const value = event.event.value;
+            if (value === DatasetAclType.Nfsv4) {
+              aclModeFormControl.setValue(AclMode.Passthrough);
+              this.entityForm.setDisabled('aclmode', false, false);
+            } else if (value === DatasetAclType.Posix) {
+              aclModeFormControl.setValue(AclMode.Discard);
+              this.entityForm.setDisabled('aclmode', true, false);
+            } else if (value === DatasetAclType.Inherit) {
+              aclModeFormControl.setValue(AclMode.Inherit);
+              this.entityForm.setDisabled('aclmode', true, false);
```
Comment (undsoft) [inline src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts:669]: When add form is opened, acltype is `inherit` by default, but `aclmode` is not disabled.

**Review by undsoft — CHANGES_REQUESTED** [review]

File: src/app/helptext/storage/volumes/datasets/dataset-form.ts [hunk 3]
```diff
@@ -40,6 +40,18 @@ export default {
...
+ written and read.\n\nWhen the ACL type is changed from POSIX to NFSv4, \
+ no migration is performed for default and access ACLs encoded in the \
+ posix1e acl extended attributes to native ZFS ACLs.\n\nWhen ACL type is \
+ changed from NFSv4 to POSIX, native ZFS ACLs are not converted to \
+ posix1e extended attributes, but the native ACL will be used internally \
+ by ZFS for access checks.\n\nThis means that the user must manually set \
+ new ACLs recursively on the dataset after ACL type changes in order to \
+ avoid unexpected permissions behavior.\n\nThis action will be destructive, \
+ and so it is advised to take a ZFS snapshot of the dataset prior to ACL \
+ type changes and permissions modifications.'),
```
Comment (undsoft) [inline src/app/helptext/storage/volumes/datasets/dataset-form.ts:53]: Sorry, can we keep the line break, but remove the empty line at least on the first two.
We need to indicate to the user that there is more text there and that he can scroll.

**Review by undsoft — APPROVED** [review]
This may be an intended behaviour, not sure.
I create a dataset with `inherit`. When I then go to edit, it won't say `inherit`, it will instead say whatever the type of the parent was, e.g. `posix`.
Looks good otherwise.

**Comment by undsoft** [comment]
Also please retarget towards `master`. We are too far in release process of 21.06 to add it in. It's going to be a known issue there.

**Comment by RehanY147** [comment]
> 
> 
> This may be an intended behaviour, not sure.
> I create a dataset with `inherit`. When I then go to edit, it won't say `inherit`, it will instead say whatever the type of the parent was, e.g. `posix`.
> Looks good otherwise.

That is intended behavior. Middleware sets the `INHERIT` value to whatever the parent`s value is.

**Comment by RehanY147** [comment]
Too many conflicts with master to change base in this PR. Will be creating a new PR.

**Pull request closed without merging**